python/packages/ai/teams/ai/models/openai_model.py

```diff
@@ -271,7 +271,34 @@ async def complete_prompt(
                     if delta.content:
                         message_content += delta.content
 
-                    # TODO: Handle tool calls
+                    # Handle tool calls during streaming
+                    if is_tools_aug and delta.tool_calls:
+                        if not hasattr(message, "action_calls") or message.action_calls is None:
+                            message.action_calls = []
+
+                        for curr_tool_call in delta.tool_calls:
+                            index = curr_tool_call.index
+
+                            # Ensure the action_calls array is long enough
+                            while index >= len(message.action_calls):
+                                message.action_calls.append(
+                                ActionCall(
+                                        id="",
+                                        function=ActionFunction(name="", arguments=""),
+                                        type="function",  # /python/packages/ai/teams/ai/prompts/message.py#L123
+                                        )
+                                )
+
+                            # Update the action_call at the specific index
+                            if curr_tool_call.id:
+                                message.action_calls[index].id = curr_tool_call.id
+                            if curr_tool_call.function:
+                                if curr_tool_call.function.name:
+                                    message.action_calls[index].function.name += curr_tool_call.function.name
+                                if curr_tool_call.function.arguments:
+                                    message.action_calls[index].function.arguments += curr_tool_call.function.arguments
+                            if curr_tool_call.type == "function":  # Type must always match the expected value
```

Comment:
extra comment

Reply:
are we supposed to set the type regardless?

Reply:
I can experiment with leaving it empty, however to my understanding if its anything other than empty, it has to be "function"